### PR TITLE
docs updates and make tools installation better

### PR DESCRIPTION
**Pull request by jaffee**
sorry about the branch name

### CODE REVIEW

**Comment by seebs** [comment]
conveniently, i was about to try to do exactly this process, so i'll look at the new instructions and follow them. will take a couple of hours probably.

File: README.md [hunk 1]
```diff
@@ -1,6 +1,29 @@
...
+   providers. This includes getting servers/vms, setting up
+   networking, firewalls, ssh access, etc.
+
+2. `ansible` configs for installing Pilosa and related utilities once
+   you have a set of servers with SSH access.
+
+You can use these tools together, or separately, and there are some
+examples of using them together along with some Makefiles to smooth
+out the whole process under `terraform/examples`. Check out the
+[readme](terraform/examples/README.org) there for more info.
```
Comment (seebs) [inline README.md]: I think README.org is no longer the right link; looks like it was renamed to README.md.

File: terraform/examples/README.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+1. Clone github.com/pilosa/infrastructure and change into it.
+2. `cd terraform/examples`
+3. Choose a cloud provider and follow the instructions in the [[../../README.md][top level README]] to get credentials set up.
+4. Copy a <cloud>-taxi-template directory to a descriptively named
+   directory. e.g. `cp -r oci-taxi-template
+   oci-taxi-DenseIO2.16-3node` Don't worry about the "taxi"
+   stuff... the Makefile includes some commands specifically for
+   running some benchmarks, but also a lot of general purpose tools.
+5. Change into that new directory. e.g. `cd oci-taxi-DenseIO2.16-3node`
+6. Copy vars.tfvars.example to vars.tfvars and change things (like the
```
Comment (seebs) [inline terraform/examples/README.md]: We only have these for some of the examples. Possibly worth fixing.

File: terraform/examples/README.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+   number of nodes, instance types, etc). Some of the clouds require
+   certain things (e.g. resource groups, network subnets, etc.) to be
+   created beforehand. If you work at Pilosa, this is all probably set
+   up, if you don't, you may have to mess around with some of the
+   terraform stuff. Feel free to get in touch, or submit an issue if
+   you have trouble.
+7. Create a new workspace: e.g. `terraform workspace new taxi-DenseIO16-3node`
+8. `make init` This sets up terraform in the local directory.
+9. `make apply` This runs terraform and actually makes your infrastructure.
+10. `make provision` This runs ansible and installs stuff on all your hosts.
```
Comment (seebs) [inline terraform/examples/README.md]: and what do i do when i want to deprovision?

File: terraform/include/Makefile [hunk 4]
```diff
@@ -27,50 +27,50 @@ require-%:
...
 		$(SSH_ARGS) \
 		$(PUBLIC_IP) "$(CMD)"
 
 provision-%: require-jq require-terraform require-ansible inventory.json  ## Provision a particular Ansible module. e.g. "make provision-agent" would run agent.yml
-	ansible-playbook -i inventory.json $(ANSIBLE_ARGS_FINAL) $(ANSIBLE_ARGS_EXTRA) $(ANSIBLE_DIR)/$*.yml
+	@ansible-playbook -i inventory.json $(ANSIBLE_ARGS_FINAL) $(ANSIBLE_ARGS_EXTRA) $(ANSIBLE_DIR)/$*.yml
 
 inventory.json: require-terraform require-jq
-	$(MAKE) output | jq '$(INVENTORY_GENERATOR_JQ)' > $@
+	@$(MAKE) output | jq '$(INVENTORY_GENERATOR_JQ)' > $@
```
Comment (seebs) [inline terraform/include/Makefile]: So, depending on the exact version of make, this sometimes  prints a "make[1] entering directory [directory it's already in]" when running this command, which goes to stdout, and thus breaks jq. With GNU make, I can `--no-print-directories` or something like that, but it seems sub-optimal. @ doesn't seem to suppress it, but I'm guessing that the --no-print-directories won't work with versions of make which don't do that...

Comment (seebs) [reply terraform/include/Makefile]:
```
 inventory.json: require-terraform require-jq
-       $(MAKE) output | jq '$(INVENTORY_GENERATOR_JQ)' > $@
+       @$(MAKE) -s output | jq '$(INVENTORY_GENERATOR_JQ)' > $@ || (echo >&2 "failing terraform output:"; $(MAKE) -s output)
```

maybe something like this?

**Comment by jaffee** [comment]
ok to merge now, I think?

**Pull request merged**